src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorTranslateDiagnosticsService.cs

```diff
@@ -162,23 +163,34 @@ private Diagnostic[] FilterCSharpDiagnostics(Diagnostic[] unmappedDiagnostics, R
     private static bool InCSharpLiteral(
         Diagnostic d,
         SourceText sourceText,
-        RazorSyntaxTree syntaxTree,
-        ILogger logger)
+        RazorSyntaxTree syntaxTree)
     {
         if (d.Range is null)
         {
             return false;
         }
 
-        var owner = syntaxTree.GetOwner(sourceText, d.Range.End, logger);
-        if (owner is null)
+        
+        var owner = syntaxTree.Root.FindNode(d.Range.AsRazorTextSpan(sourceText), getInnermostNodeForTie: true);
+        if (IsCsharpKind(owner))
         {
-            return false;
+            return true;
         }
 
-        var isCSharp = owner.Kind is SyntaxKind.CSharpExpressionLiteral or SyntaxKind.CSharpStatementLiteral or SyntaxKind.CSharpEphemeralTextLiteral;
+        if (owner is CSharpImplicitExpressionSyntax implicitExpressionSyntax
+            && implicitExpressionSyntax.Body is CSharpImplicitExpressionBodySyntax bodySyntax
+            && bodySyntax.CSharpCode is CSharpCodeBlockSyntax codeBlock)
+        {
+            return codeBlock.Children.Count == 1
+                && IsCsharpKind(codeBlock.Children[0]);
+        }
```

Comment:
I think this whole thing is possibly:

```
        return (owner is CSharpImplicitExpressionSyntax 
            {
                Body: CSharpImplicitExpressionBodySyntax
                {
                    CSharpCode: CSharpCodeBlockSyntax
                    {
                        Children: [var firstChild]
                    }
                }
            } &&
            IsCsharpKind(firstChild);
```

though I'm not sure thats actually readable in any way, so this is not a suggestion to change it necessarily, I just enjoy these trivia challenges.

Also since when did the `&&` go at the start of the line? I always put them at the end, so the conditions all line up neatly under the `if (`

Reply:
> Also since when did the && go at the start of the line? 

This was from using the `Wrap Expression` code fix. I'm happy with either way though. 

Benefit of this style is for diffing, so you only see the line being added/removed, and don't mess with others. Similar to wrapping arguments with 

```csharp
static void Method(
	arg1
	, arg2
	, arg3
```
 

Although for methods and enums I'd prefer to just leave a trailing comma if possible

Reply:
"Similar to <other convention I find even more ugly>" is not the argument you think it is 😛

Reply:
Personally, I prefer `&&` at the start of the line because I find it more readable. I'm only wrapping when the expression is long, and the `&&` is more visible at the start of the line, which (imo) makes the precedence stand out more.

Reply:
![WannaFightJackReacherGIF](https://github.com/dotnet/razor/assets/754264/b271e15e-28cb-4f32-b44e-5b01b430b9f5)